src/Nethermind/Nethermind.Mev/Data/MevBundle.cs

```diff
@@ -18,21 +18,31 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Threading;
 using Nethermind.Core;
 using Nethermind.Core.Crypto;
 using Nethermind.Int256;
+using Nethermind.Mev.Source;
+using Nethermind.Serialization.Rlp;
 
 namespace Nethermind.Mev.Data
 {
-    public class MevBundle : IEquatable<MevBundle>
+    public class MevBundle : IEquatable<MevBundle>, IComparable<MevBundle>
     {
+        private static int _poolIndex = 0;
+
         public MevBundle(IReadOnlyList<Transaction> transactions, long blockNumber, UInt256? minTimestamp, UInt256? maxTimestamp, Keccak[]? revertingTxHashes = null)
         {
             Transactions = transactions;
             BlockNumber = blockNumber;
+
+            Rlp rlp = Rlp.Encode(Rlp.Encode(BlockNumber), Rlp.Encode(transactions.Select(t => t.Hash).ToArray()));
+            Hash = Keccak.Compute(rlp.Bytes);
+            
             RevertingTxHashes = revertingTxHashes ?? Array.Empty<Keccak>();
             MinTimestamp = minTimestamp ?? UInt256.Zero;
             MaxTimestamp = maxTimestamp ?? UInt256.Zero;
+            PoolIndex = Interlocked.Increment(ref _poolIndex);
```

Comment:
SequenceNumber / Id

Reply:
What is it needed for if we have Hash?

Reply:
If 2 bundles with same identity come (same blocknumber and same transactions in same order), we want to keep the last one that came (they can differ in timestamps)